packaging_automation/citus_package.py

```diff
@@ -264,31 +284,38 @@ def get_docker_image_name(platform: str):
 @validate_parameters
 # disabled since this is related to parameter_validations library methods
 # pylint: disable=no-value-for-parameter
+# pylint: disable= too-many-locals
 def build_packages(github_token: non_empty(non_blank(str)),
                    platform: non_empty(non_blank(str)),
                    build_type: BuildType, signing_credentials: SigningCredentials,
                    input_output_parameters: InputOutputParameters, is_test: bool = False) -> None:
     os_name, os_version = decode_os_and_release(platform)
-    release_versions, nightly_versions = get_postgres_versions(os_name, input_output_parameters.input_files_dir)
+    release_versions, nightly_versions = get_postgres_versions(platform,
+                                                               input_output_parameters.input_files_dir)
+
     signing_credentials = get_signing_credentials(signing_credentials.secret_key, signing_credentials.passphrase)
+
     if platform != "pgxn":
         package_version = get_package_version_without_release_stage_from_pkgvars(
             input_output_parameters.input_files_dir)
-        write_postgres_versions_into_file(input_output_parameters.input_files_dir, package_version)
+        write_postgres_versions_into_file(input_output_parameters.input_files_dir, package_version, os_name, platform)
 
     if not signing_credentials.passphrase:
         raise ValueError("PACKAGING_PASSPHRASE should not be null or empty")
+    postgress_versions_to_process = release_versions if build_type == BuildType.release else nightly_versions
+
+    postgres_docker_extension_iterator = ["all"] if platform_postgres_version_source[
+                                                        os_name] == PostgresVersionDockerImageType.single \
+        else postgress_versions_to_process
```

Comment:
This is a bit too hard to read. Can we rewrite the same logic?

```suggestion
    if platform_postgres_version_source[os_name] == PostgresVersionDockerImageType.single :
        postgres_docker_extension_iterator = ["all"] 
    else :
        postgres_docker_extension_iterator = postgress_versions_to_process
```

Reply:
I see that you don't like one liner if checks. Actually it looks me clearer but no problem we can change it